### PR TITLE
unit tests for Transactions and GetByTransliterationUrl

### PR DESCRIPTION
**Pull request by ormykhalyshyn**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/GetStreetcodeByTransliterationUrlHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+            //Act
+            var result = await handler.Handle(new GetStreetcodeByTransliterationUrlQuery(url), CancellationToken.None);
+
+            //Assert
+            Assert.Multiple(
+                () => Assert.NotNull(result),
+                () => Assert.True(result.IsSuccess),
+                () => Assert.Equal(result.Value.TransliterationUrl, url)
+            );
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/GetStreetcodeByTransliterationUrlHandlerTests.cs:70]: if you have too many assertions, consider moving it in the separate tests.

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/GetStreetcodeByTransliterationUrlHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                () => Assert.Equal(expectedError, result.Errors.First().Message)
+            );
+
+        }
+
+        [Theory]
+        [InlineData("some")]
+        public async Task CorrectType(string url)
+        {
+
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/GetStreetcodeByTransliterationUrlHandlerTests.cs:106]: same here

File: Streetcode/Streetcode.XUnitTest/MediatRTests/Transactions/TransactionsTests/TransactionLinkTests/GetAllTransactLinksHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    public class GetAllTransactLinksHandlerTests
+    {
+        private readonly Mock<IRepositoryWrapper> _mockRepo;
+        private readonly Mock<IMapper> _mockMapper;
+        public GetAllTransactLinksHandlerTests()
+        {
+            _mockMapper = new Mock<IMapper>();
+            _mockRepo = new Mock<IRepositoryWrapper>();
+        }
+        async Task SetupRepository(List<TransactionLink> returnList)
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/Transactions/TransactionsTests/TransactionLinkTests/GetAllTransactLinksHandlerTests.cs:31]: is it useful to use async methods for mock setups?

File: Streetcode/Streetcode.BLL/DTO/Streetcode/StreetcodeDTO.cs [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 namespace Streetcode.BLL.DTO.Streetcode;
 
-public abstract class StreetcodeDTO
+public /*abstract*/ class StreetcodeDTO
```
Comment (Kasterov) [inline Streetcode/Streetcode.BLL/DTO/Streetcode/StreetcodeDTO.cs:6]: Why this class was abstract? 😅 And why this line is comented now?

File: Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/GetStreetcodeByTransliterationUrlHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        }
+
+        private static StreetcodeDTO returnNullDTO(string url)
+        {
+
+            return null;
+
+        }
+
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/Streetcode/GetStreetcodeByTransliterationUrlHandlerTests.cs:137]: same here with returning nulls.

File: Streetcode/Streetcode.XUnitTest/MediatRTests/AdditionalContent/AdditionalContentTests/TagTests/CreateTagHandlerTests.cs [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Tysyatsky) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/AdditionalContent/AdditionalContentTests/TagTests/CreateTagHandlerTests.cs:117]: same issue with spaces

**Review by Tysyatsky — COMMENTED** [review]
Great work! Added comments for overall code style and some other changes.

File: Streetcode/Streetcode.XUnitTest/MediatRTests/AdditionalContent/AdditionalContentTests/TagTests/CreateTagHandlerTests.cs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+            return new CreateTagDTO
+            {
+                Title = GetTag().Title
+            };
+        }
+
+
+    }
+}
+
```
Comment (Kasterov) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/AdditionalContent/AdditionalContentTests/TagTests/CreateTagHandlerTests.cs:98]: A lot of empty lines. Delete them, please.

File: Streetcode/Streetcode.XUnitTest/MediatRTests/AdditionalContent/AdditionalContentTests/TagTests/CreateTagHandlerTests.cs [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+        {
+
+            // Arrange
+
+            await SetupRepository(GetTag());
+            await SetupMapper(GetTag());
+
+            var exception = new Exception("Database error.");
+
+            _mockRepo.Setup(r => r.TagRepository.CreateAsync(GetTag())).ThrowsAsync(exception);
```
Comment (Kasterov) [inline Streetcode/Streetcode.XUnitTest/MediatRTests/AdditionalContent/AdditionalContentTests/TagTests/CreateTagHandlerTests.cs:66]: This part of code look a litle bit confusing. ThrowsAsync should be used in Assert part of UnitTest not in Arrange (I assume taking into consideration StackOverflow and other articles devoted to that theme). Also, do you need this throwing here? Current test called ShouldReturnSuccessfully_TypeIsCorrect and nothing about exeption handling... Maybe I did not catch that at all.

**Review by Kasterov — CHANGES_REQUESTED** [review]
Well done, but need to be a little bit refactored!

**Review by Tysyatsky — APPROVED** [review]

**Review by Kasterov — APPROVED** [review]
Best UnitTest I have ever seen! Well done!

**Pull request merged**